### PR TITLE
Added drop variables feature to Stack(Pivot Longer) dialog

### LINKED ISSUE #6868: Small addition to the new stack (pivot longer) dialogue
@shadrackkibet this is a good simple task for an intern.

I have been going through some examples for this dialogue from the pivotting vignette for the tidyr package.

The billboard data (from the tidyr package) includes the option from the command `values_drop_na = TRUE`.

Could there be a checkbox on the left, default unchecked, with label Drop Missing Values.

### PR DESCRIPTION
**Pull request by Vitalis95**
Fixes #6868 
@rdstern @lloyddewit @N-thony , added `Drop Variables` control  which drops the variables  before reshaping.

### CODE REVIEW

**Review by rdstern — APPROVED** [review]
Looks fine now

File: instat/dlgStack.vb [hunk 1]
```diff
@@ -302,6 +323,15 @@ Public Class dlgStack
             Else
                 ucrBase.OKEnabled(True)
             End If
+            If ucrChkStackMultipleSets.Checked Then
+                If ucrChkDropVariables.Checked AndAlso ucrReceiverDropValues.IsEmpty Then
+                    ucrBase.OKEnabled(False)
+
+                Else
+                    ucrBase.OKEnabled(True)
+
```
Comment (lloyddewit) [inline instat/dlgStack.vb:332]:
```suggestion
```

File: instat/dlgStack.vb [hunk 2]
```diff
@@ -396,15 +426,28 @@ Public Class dlgStack
...
                                              , clsReshapeFunction, clsPivotLongerFunction))
         If ucrChkStackMultipleSets.Checked Then
+            ucrChkDropPrefix.Visible = False
             ucrReceiverColumnsToBeStack.SetMeAsReceiver()
+            If ucrChkDropVariables.Checked Then
+                ucrReceiverDropValues.SetMeAsReceiver()
+            End If
+        Else
+            ucrChkDropPrefix.Visible = True
+
```
Comment (lloyddewit) [inline instat/dlgStack.vb:445]:
```suggestion
```

File: instat/dlgStack.vb [hunk 3]
```diff
@@ -429,4 +472,28 @@ Public Class dlgStack
             ucrReceiverExpand.SetMeAsReceiver()
         End If
     End Sub
+
+    Private Sub ucrChkDropVariables_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrChkDropVariables.ControlValueChanged
+        If ucrChkDropVariables.Checked Then
+            clsDummyFunction.AddParameter("drop", "True", iPosition:=0)
+        Else
+            clsDummyFunction.AddParameter("drop", "False", iPosition:=0)
+
```
Comment (lloyddewit) [inline instat/dlgStack.vb:481]:
```suggestion
```

**Review by lloyddewit — APPROVED** [review]

**Pull request merged**